app/src/composables/use-item.ts

```diff
@@ -172,9 +172,15 @@ export function useItem(
 			...edits.value,
 		};
 
-		// Make sure to delete the primary key if it's generated
-		if (primaryKeyField.value && primaryKeyField.value.schema?.is_generated && primaryKeyField.value.field in newItem) {
-			delete newItem[primaryKeyField.value.field];
+		// Make sure to delete the primary key if it's has auto increment enabled
+		if (primaryKeyField.value && primaryKeyField.value.field in newItem) {
+			if (
+				primaryKeyField.value.schema?.has_auto_increment ||
+				primaryKeyField.value?.schema?.is_generated ||
```

Comment:
```suggestion
				primaryKeyField.value.schema?.is_generated ||
```

Reply:
@azrikahar Im starting to doubt the function of `is_generated` 🤔 in my tests none of the 3 primary key types get the `is_generated` flag set to true. Am i missing a usecase?

Reply:
That is a very good question 🤔 It seems like it was added in #14947, do you recall the context of it? I'm also not sure the usage of `is_generated` in that PR, or maybe the fix actually required a different approach.

Reply:
This was probably a mistaken assumption on my part on what `is_generated` was intended for 😬 It doesn't look to be a flag related to primary keys 🤦

Reply:
Honestly I've also never 100% grasped it's purpose until now. Seems like it is indeed meant for generated columns rather than primary keys specifically.

Seems like the original `primaryKeyField.value && primaryKeyField.value.field in newItem` would still be the correct logic, but now we need to prevent this primary key deletion for manually entered primary keys.

What if we first delete the primary key from the fetched data first (or somehow _not_ fetch primary key), then merge `edits.value` into it? Not sure if there's other edge cases not covered by this though.

Reply:
I think this check should cover it https://github.com/directus/directus/pull/16966/files#diff-4ca9afb518ec3f7957e6a2723f01f557c177987383e93bda1ce02918b73b27fcR177
As manually entered IDs neither have `is_auto_increment` set not have the `uuid` special.

Reply:
Looked into it a little and the ID is either needed for the update request or it needs to be there as value on the edits (like manually entered ids).